frontend/test/pytest/test_transform.py

```diff
@@ -0,0 +1,287 @@
+            qml.RY(0.5, wires=1)
+            qml.RX(1.3, wires=2)
+
+            qml.CNOT(wires=[0, 1])
+            qml.WireCut(wires=1)
+            qml.CNOT(wires=[1, 2])
+
+            qml.RX(x, wires=0)
+            qml.RY(0.7, wires=1)
+            qml.RX(2.3, wires=2)
+            return qml.sample(wires=[0, 2])
+
+        v = 0.319
+        target = circuit(v)
+
+        cut_circuit_bs = jax.jit(qcut.cut_circuit_mc(circuit))
+        cut_res_bs = cut_circuit_bs(v)
+
+        assert cut_res_bs.shape == target.shape
+        assert isinstance(cut_res_bs, type(target))
+
+        cut_circuit_bs_compiled = qjit(qcut.cut_circuit_mc(circuit))
+        cut_res_bs_compiled = cut_circuit_bs_compiled(v)
+
+        assert cut_res_bs_compiled.shape == target.shape
+        assert isinstance(cut_res_bs_compiled, type(target))
+        """
+
+        dev = qml.device("lightning.qubit", wires=2, shots=None)
+
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.RX(x, wires=0)
+            qml.RY(0.543, wires=1)
+            qml.WireCut(wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.RZ(0.240, wires=0)
+            qml.RZ(0.133, wires=1)
+            return qml.expval(qml.PauliZ(wires=[0]))
+
+        x = jnp.array(0.531)
+        cut_circuit_jit = jax.jit(qcut.cut_circuit(circuit, use_opt_einsum=False))
+        cut_circuit_qjit = qjit(qcut.cut_circuit(circuit, use_opt_einsum=False))
+
+        # Note we call the function twice but assert qcut_processing_fn is called once. We expect
+        # qcut_processing_fn to be called once during JIT compilation, with subsequent calls to
+        # cut_circuit_jit using the compiled code.
+        res = cut_circuit_jit(x)
+        res_qjit = cut_circuit_qjit(x)
+        res_expected = circuit(x)
+
+        assert np.allclose(res_qjit, res_expected)
+        assert np.allclose(res, res_expected)
+
+class TestHamiltonianExpand:
+
+    def test_hamiltonian_expand(self):
+        H4 = (qml.PauliX(0) @ qml.PauliZ(2)
+                + 3 * qml.PauliZ(2)
+                - 2 * qml.PauliX(0)
+                + qml.PauliZ(2)
+                + qml.PauliZ(2)
+        )
+        H4 += qml.PauliZ(0) @ qml.PauliX(1) @ qml.PauliY(2)
+
+        dev = qml.device("lightning.qubit", wires=3)
+
+        @hamiltonian_expand
+        @qml.qnode(dev)
+        def circuit():
+            qml.Hadamard(0)
+            qml.Hadamard(1)
+            qml.PauliZ(1)
+            qml.PauliX(2)
+            return qml.expval(H4)
+
+        assert np.allclose(circuit(), qjit(circuit)())
+
+class TestSumExpand:
+
+    def test_sum_expand(self):
+        dev = qml.device("lightning.qubit", wires=2, shots=None)
+
+        @sum_expand
+        @qml.qnode(dev)
+        def circuit():
+            obs1 = qml.prod(qml.PauliX(0), qml.PauliX(1))
+            obs2 = qml.prod(qml.PauliX(0), qml.PauliY(1))
+            return [qml.expval(obs1), qml.expval(obs2)]
+
+        assert np.allclose(circuit(), qjit(circuit)())
```

Comment:
@erick-xanadu might be good to also test some other transforms that simply go tape -> tape, like `merge_rotations`!

Reply:
Added test for merge rotations: https://github.com/PennyLaneAI/catalyst/pull/280/commits/c3d8a5c00c7dc7e064edada3a460d4408f22a345